inc/block-editor.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * Block editor (gutenberg) specific functionality
+ *
+ * @package UnderStrap
+ */
+
+
+add_action( 'after_setup_theme', 'understrap_block_editor_setup' );
+
+if ( ! function_exists( 'understrap_block_editor_setup' ) ) {
+
+	/**
+	 * Sets up our default theme support for the WordPress block editor.
+	 *
+	 */
+	function understrap_block_editor_setup() {
+
+		// Add support for the block editor stylesheet.
+		add_theme_support( 'editor-styles' );
+
+		// Add support for wide alignment.
+		add_theme_support( 'align-wide' );
+
+		// Register our custom colors as options in the editor.
+		$color_palette = understrap_generate_color_palette();
+		if ( $color_palette ) {
+			add_theme_support( 'editor-color-palette', $color_palette );
+		}
+	}
+}
+
+if ( ! function_exists( 'understrap_generate_color_palette' ) ) {
+	/**
+	 * Checks for our JSON file of color values. If exists, creates a color palette array.
+	 *
+	 * @return array|bool
+	 */
+	function understrap_generate_color_palette() {
+		$color_palette = array();
+
+		// Grabs the autogenerated color palette that we're pulling from our compiled bootstrap stylesheets.
+		$color_palette_json = wp_remote_get( get_stylesheet_directory_uri() . '/inc/editor-color-palette.json' );
+		if ( ! is_wp_error( $color_palette_json ) ) {
+			$color_palette_json = json_decode( $color_palette_json['body'], true );
+			foreach ( $color_palette_json as $key => $value ) {
+				$key             = ltrim( $key, '--' );
+				$color_palette[] = array(
+					'name'  => $key,
```

Comment:
The color name is supposed to be a translatable string. Non-English speaking screen reader users won't know which color it is unless you provide a translation. This is a no go in terms of accessibility.

Reply:
Interesting. In that case - I see no reason not to skip the loop function and filter overall. I can hard code the entire thing with the default list of colors in bootstrap (blue, red, etc) while still pulling the hex 'value' from the json file. When I generate the list in gulpconfig.js I specify the colors there, so the precedent exists.